src/test/java/edu/ucla/library/prl/harvester/JobRequestsFT.java

```diff
@@ -0,0 +1,204 @@
+
+package edu.ucla.library.prl.harvester;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.net.MalformedURLException;
+import java.text.ParseException;
+
+import javax.mail.internet.AddressException;
+
+import org.apache.http.HttpStatus;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import com.google.i18n.phonenumbers.NumberParseException;
+
+import edu.ucla.library.prl.harvester.services.HarvestScheduleStoreService;
+import edu.ucla.library.prl.harvester.utils.TestUtils;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import io.vertx.config.ConfigRetriever;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.sqlclient.Pool;
+import io.vertx.uritemplate.UriTemplate;
+
+/**
+ * Tests the application's behavior in response to requests involving {@link Job}s.
+ */
+@ExtendWith(VertxExtension.class)
+@TestInstance(Lifecycle.PER_CLASS)
+public class JobRequestsFT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JobRequestsFT.class, MessageCodes.BUNDLE);
+
+    private static final UriTemplate INSTITUTIONS = UriTemplate.of("/institutions");
+
+    private static final UriTemplate JOB = UriTemplate.of("/jobs/{id}");
+
+    private static final UriTemplate JOBS = UriTemplate.of("/jobs");
+
+    private Pool myDbConnectionPool;
+
+    private WebClient myWebClient;
+
+    private int myInstitutionID;
+
+    /**
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @BeforeAll
+    public final void setUp(final Vertx aVertx, final VertxTestContext aContext) {
+        ConfigRetriever.create(aVertx).getConfig().compose(config -> {
+            final String host = config.getString(Config.HTTP_HOST);
+            final int port = config.getInteger(Config.HTTP_PORT);
+
+            myDbConnectionPool = HarvestScheduleStoreService.getConnectionPool(aVertx, config);
+            myWebClient = WebClient.create(aVertx, new WebClientOptions().setDefaultHost(host).setDefaultPort(port));
+
+            return Future.succeededFuture();
+        }).onSuccess(result -> aContext.completeNow()).onFailure(aContext::failNow);
+    }
+
+    /**
```

Comment:
Even on tests, I'm pro-javadocs. Even if you skip the description of the test, putting the `@throws AddressException` without documentation of when/why this happens is just skirting the rules which are requiring that they be there. If that's the route you want to go, I'd suggest changing the checkstyle (or PMD?) rule that's requiring javadocs on the tests (that seems better to me than doing the bare minimum just to get the rules to pass).

Reply:
For the record, so am I. The PMD and Checkstyle configurations that were inherited from your template project don't actually apply the rules to `src/test` (which I assumed was the intended behavior), which is why there are examples like [this](https://github.com/UCLALibrary/prl-harvester/blob/98575f759e883ff44ba4e9e4187b85f383c26aa8/src/test/java/edu/ucla/library/prl/harvester/utils/TestUtils.java#L80-L84) which doesn't even list the thrown exceptions, as well as in other projects and in other ways (undocumented member variables being common). This also isn't the first PR of mine on this project where I've left off descriptions of why particular exceptions are thrown by a test method. Just saying; I'm happy to follow the same rules in `src/test` as we do in `src/main`, but some consistency would help my understanding of expectations.

Reply:
Hmm, thanks for that [this](https://github.com/UCLALibrary/prl-harvester/blob/98575f759e883ff44ba4e9e4187b85f383c26aa8/src/test/java/edu/ucla/library/prl/harvester/utils/TestUtils.java#L80-L84)... that looks like a bug in the upstream rules. My take is every param, return value, throws, etc. should have documentation (that it's better to err on the side of too much/redundant documentation than not enough). I knew it was possible to not actually include text if you included the tag, but didn't know it was possible to skip the throws tag altogether. I'll create a bug ticket on that project, thanks. It's also my desire to keep improving the rules so that they require actual text after the tag, but that's obviously not something I've gotten to yet.

Reply:
As for expectations, there is no requirement that this project has to follow the upstream parent project's rules, as we've already overridden one naming pattern. That's up to you.